# HG changeset patch
# User Petri Hintukainen <phintuka@...>
# Date 1339310702 -10800
# Node ID b0df4b6003ad2b7454059c2d82ef7f8f29e48e9e
# Branch  default
# Parent  52a99e60c2c9aa7e03be894b4396f229f881025c
Fix building on non-x86 systems

diff --git a/src/post/deinterlace/plugins/greedy2frame_template.c b/src/post/deinterlace/plugins/greedy2frame_template.c
--- a/src/post/deinterlace/plugins/greedy2frame_template.c
+++ b/src/post/deinterlace/plugins/greedy2frame_template.c
@@ -85,6 +85,7 @@
 */
 
 
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
 #if !defined(MASKS_DEFINED)
 #define MASKS_DEFINED
 static const mmx_t Mask = { uq: 0x7f7f7f7f7f7f7f7fll };
@@ -92,6 +93,7 @@
 static const mmx_t GreedyTwoFrameThreshold = { ub: {TP, TP, TP, TP} };
 #undef TP
 #endif
+#endif
 
 #if defined(IS_MMXEXT)
 static void DeinterlaceGreedy2Frame_MMXEXT(uint8_t *output, int outstride,
 --git a/src/post/deinterlace/plugins/greedy2frame_template_sse2.c b/src/post/deinterlace/plugins/greedy2frame_template_sse2.c
--- a/src/post/deinterlace/plugins/greedy2frame_template_sse2.c
+++ b/src/post/deinterlace/plugins/greedy2frame_template_sse2.c
@@ -84,11 +84,12 @@
 **   B0    |         |    B1   |         |
 */
 
-
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
 static const sse_t Mask128 = { uq: { 0x7f7f7f7f7f7f7f7fll, 0x7f7f7f7f7f7f7f7fll} };
 #define TP GREEDYTWOFRAMETHRESHOLD, GREEDYTWOFRAMETHRESHOLD2
 static const sse_t GreedyTwoFrameThreshold128 = { ub: {TP, TP, TP, TP, TP, TP, TP, TP} };
 #undef TP
+#endif
 
 static void DeinterlaceGreedy2Frame_SSE2(uint8_t *output, int outstride,
                                          deinterlace_frame_data_t *data,


